### DISCUSSION TITLE
Booklet flip side behaviour

### DISCUSSION THREAD

**Discussion by nicolasfranck** [pdfcpu/pdfcpu#824] [Q&A]
I started testing the booklet command (in two-up), and noticed that the back side is already
flipped in the pdf. Isn't that something for the printer to do? When I print the
pdf with option two-sided on, the back side is turned .. on its back.

Also, is there a reason why the pages are arranged on a portrait page (in which the original pages
are put on a landscape with rotation 90%), instead of on a landscape with the pages
horizontally side by side?

**Comment by hhrutter** [comment]
@adamgreenhall can you help out?

**Comment by adamgreenhall** [comment]
@nicolasfranck good questions. 

For the first one, *some* printers do duplexing automatically. But the printer I use does not. Adding an option for how the duplexing flip  should work (auto/long-edge/short-edge) would make sense. For reference, the MacOS printer dialog looks something like this:
<img width="625" alt="Screenshot 2024-03-08 at 4 53 00 PM" src="https://github.com/pdfcpu/pdfcpu/assets/704919/6882efa2-38ab-49cf-b641-06bc287653bc">

For your second question, I think it makes sense to keep the 2-up output pdf orientation portrait, since that is the way things get sent to the printer. Having said that, currently NUp creates an output pdf sheet with the same orientation as the input pdf (and Booklet also does this). So if you send something portrait, you'll get a portrait pdf out - and the same for landscape.

**Comment by nicolasfranck** [comment]
Thanks for your quick reply!

So if I understand correctly, the back sides are already flipped because it is intended for printers were duplexing is not done automatically; and so people have to manually flip the pages one by one, and feed them back into the printer in order to proceed the printing process. And the way of turning is already laid out in the digital form. That manual flipping sounds like a lot of manual and error prone work, not? Especially if it is not a regular page ordering (with one digital page on a physical page), but one that is rearranged for a booklet binding.

It would indeed be a good option to make this behaviour configurable, somewhere in the future ;-)

About the portrait layout: it is indeed true that the printer has only one way of printing (portrait), and that landscape printing is just the layout rotated by 90% and put on a portrait. It is just harder to validate the input pdf as a human, and the printer eventually prints landscape pdf pages on a portrait page anyway.

btw. thanks for this great tool! This is not meant as any sort of critique. I am merely trying to figure out why certain decisions were made

**Comment by adamgreenhall** [comment]
Sorry - on the first question, I got confused and answered incorrectly. The main correction to takeaway is that if you print a 2up booklet with a normal printer, it will work just fine. 

So why are the 2up booklet pages rotated 180deg? When printers do double sided printing (or if you do it manually), they flip the paper along the short side axis and then print again on the back. If you think about a normal (non-booklet) document that you print two sided, the second page will have the left-most text printed on the other side of the paper as the right-most text from the first page. 

But -- for booklets, we want the even pages to be paired to the back of their corresponding odd page - and facing the same direction. This pairing is easier to see in 4up. Here are the first two sheets - arranged so that pg2 will wind up on the back of pg2 (when the sheet is flipped during printing):
<img width="120" alt="Screenshot 2024-03-13 at 8 23 21 AM" src="https://github.com/pdfcpu/pdfcpu/assets/704919/794be72b-229c-4e00-9bc6-b1cde53dd788">

For 2up, because the pages are rotated to fit on the sheet, we need to also rotate the even pages 180deg to get them to face the same direction when the sheet is flipped during printing. Here are the first two sheets - and the even pages are rotated, so that the top of pg2 to face the same direction as the top of pg1 when printed.
<img width="121" alt="Screenshot 2024-03-13 at 8 27 15 AM" src="https://github.com/pdfcpu/pdfcpu/assets/704919/8f365459-9559-4f37-a365-3392b5d50cee">

**Comment by nicolasfranck** [comment]
Mm, I tried with this pdf

[word_booklet.pdf](https://github.com/pdfcpu/pdfcpu/files/14598759/word_booklet.pdf)

and with printer settings "double sided (short-edge)" and "portrait" it came as follows (front and back):

front: 
![IMG_1522](https://github.com/pdfcpu/pdfcpu/assets/494636/aa4b6224-d089-4a9e-9baa-67c967ac5f6e)

back:
![IMG_1523](https://github.com/pdfcpu/pdfcpu/assets/494636/c945d923-02d7-479d-ac72-0517428a556d)

Which sounds logic to me: if it is flipped around the short edge, the "1" and "7" are close to each other.

**Comment by hhrutter** [comment]
closed due to inactivity
